vcpkg/consume/lock-package-versions.md

```diff
@@ -0,0 +1,238 @@
+> * [Create a project with a manifest]
+> * [Add version constraints using a baseline]
+> * [Update the baseline versions]
+> * [Add a minimum version constraint]
+> * [Force a specific version]
+
+## Prerequisites
+
+* A terminal
+* A code editor
+* vcpkg
+* CMake
+
+## 1 - Create a project with a manifest
+
+In an empty folder, create the following project files:
+
+A source file (`main.cpp`):
+
+:::code language="cpp" source="snippets/versioning-main.cpp":::
+
+A CMake project file (`CMakeLists.txt`):
+
+:::code language="cmake" source="snippets/versioning-cmakelists.txt":::
+
+A vcpkg manifest (`vcpkg.json`):
+
+```json
+{
+  "dependencies": [ "fmt", "zlib" ]
+}
+```
+
+Build the project, replace `%VCPKG_ROOT%` with your vcpkg installation path:
+
+```bash
+cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=%VCPKG_ROOT%/scripts/buildsystems/vcpkg.cmake
+cmake --build build
+```
+
+Run the program:
+
+```Console
+fmt version is 70103
+zlib version is 1.2.11
+```
+
+It is likely, that when you run the program the versions of these libraries are different than the
+output above. In the next step, we show you how to pin the versions of these dependencies so that
+they stay consistent each time you build the project.
+
+## 2 - Add version constraints using a baseline
+
+A version baseline, establishes a minimum version floor for all the packages. Read the [vcpkg
+concepts](../users/versioning.md#baselines) to learn about baselines.
+
+To get the exact versions used in the previous step, modify the contents of `vcpkg.json` to:
+
+```json
+{
+  "dependencies": [
+    "fmt",
+    "zlib"
+  ],
+  "builtin-baseline": "3426db05b996481ca31e95fff3734cf23e0f51bc"
+}
+```
+
+Setting [`builtin-baseline`](../reference/vcpkg-json.md#builtin-baseline) to a specific commit SHA
+of the vcpkg repository instructs vcpkg to use the package versions at that specific commit as the minimum
+version for all the packages.
+
+You can use Git to examine the versions for that particular baseline:
+
+```bash
+git show 3426db05b996481ca31e95fff3734cf23e0f51bc:versions/baseline.json | egrep -A 3 -e '"zlib"|"fmt"'
+    "fmt": {
+      "baseline": "7.1.3",
+      "port-version": 1
+    },
+--
+    "zlib": {
+      "baseline": "1.2.11",
+      "port-version": 9
+    },
+```
+
+## 3 - Update the baseline versions
+
+Baselines offer a convenient mechanism to update the versions of all your dependencies at once. To
+update your baseline run the following command:
+
+```bash
+vcpkg x-update-baseline
+```
+
+The [`x-update-baseline`](../commands/update-baseline.md) command modifies your manifest file to set
+[`builtin-baseline`](../reference/vcpkg-json.md#builtin-baseline) to the current Git commit of your
+vcpkg instance.
+
+You can use the `--add-initial-baseline` option to add a `builtin-baseline` to a manifest that does
+not have one yet.
+
+## 4 - Add a minimum version constraint
+
+Baselines are not the only way to lock down a package's version. vcpkg also accepts minimum version
```

Comment:
I know you describe it below but it would be good to add an inserted "important" block or similar at the top of this chunk saying something like:

> Important: Adding version constraints will only choose versions later than those in your baseline. To choose older versions, see overrides below.

I know you describe that in terms of 'how it works' here, and that is good, but I think calling out the 'so what factor' for like 80% of customers is good to do early in the section if possible.